esx_service/vmdk_ops.py

```diff
@@ -754,11 +754,25 @@ def get_datastore_url_from_config_path(config_path):
                   % (config_path, config_ds_url))
     return config_ds_url
 
-def datastore_path_exist(datastore):
-    """ Check whether path /vmfs/volumes/datastore" exist or not """
-    ds_path = os.path.join("/vmfs/volumes/", datastore)
+def datastore_path_exist(datastore_name):
+    """ Check whether path /vmfs/volumes/datastore_name" exist or not """
+    ds_path = os.path.join("/vmfs/volumes/", datastore_name)
     return os.path.exists(ds_path)
 
+def get_datastore_name(datastore_url):
+    """ Get datastore_name with given datastore_url """
+    datastore_name = vmdk_utils.get_datastore_name(datastore_url)
+    datastore_path = os.path.join("/vmfs/volumes/", datastore_name)
+    if not datastore_path_exist(datastore_path):
```

Comment:
Shouldn't we pass the datastore_name instead of datastore_path here? See the API signature at line 757.